app/routes/post_login.js

```diff
@@ -38,15 +38,36 @@ export default Ember.Route.extend(preloadDataMixin, {
         new AjaxPromise(`/orders/${draftOrder.id}`, "PUT", this.get('session.authToken'), { order: orderParams })
           .then(data => {
             this.get("store").pushPayload(data);
+            this.redirectToTransitionOrDetails();
           });
       }
+    }else{
+      this.redirectToTransitionOrDetails();
     }
+  },
+
+  redirectToTransitionOrDetails() {
+    if(this.isDetailsComplete()){
+      var attemptedTransition = this.controllerFor('login').get('attemptedTransition');
+      if (attemptedTransition) {
+        attemptedTransition.retry();
+        this.controllerFor('login').set('attemptedTransition', null);
+      }else{
+        this.transitionTo("browse");
+      }
+    }else{
+      this.transitionTo("account_details");
+    }
+  },
 
-    // After everthying has been loaded, redirect user to requested url
-    var attemptedTransition = this.controllerFor('login').get('attemptedTransition');
-    if (attemptedTransition) {
-      attemptedTransition.retry();
-      this.set('attemptedTransition', null);
+  isDetailsComplete(){
+    var organisation = this.store.peekAll('organisation').objectAt(0);
```

Comment:
How do you know that it'll return the `organisation/organisationsUser/user` that you need? Because you're getting all records from ember-data and taking first object. 
The OrganisationUser you get might not be related to that Organisation at all.


CC: @patrixr , @namrataukirde